src/awkward/_v2/contents/content.py

```diff
@@ -265,7 +282,11 @@ def __getitem__(self, where):
                 return self.__getitem__((where,))
 
             elif isinstance(where, Iterable) and all(ak._util.isstr(x) for x in where):
-                return self._getitem_fields(where)
+                if len(where) == 0:
+                    # FIXME
+                    return []
```

Comment:
This is related to the comment on line 224 of this file. `Content.__getitem__` definitely should not be returning a Python list (the allowed return types are `Content`, `dtype.type` (i.e. int, bool, ...), and `None`). The `_getitem_fields` method should be handling an empty `where` by converting the first RecordArray into a RecordArray with no fields. If it's not, that's where it should be fixed.

RecordArrays can have zero `contents`; this is why they need a `length` argument:

```python
>>> RecordArray([], [], 10)
<RecordArray len='10' is_tuple='False'>
</RecordArray>
>>> ak.to_list(RecordArray([], [], 10))
[{}, {}, {}, {}, {}, {}, {}, {}, {}, {}]
```